openflow13/openflow13.go

```diff
@@ -349,6 +349,9 @@ func (p *PacketIn) UnmarshalBinary(data []byte) error {
 	n += 8
 
 	err = p.Match.UnmarshalBinary(data[n:])
+	if err != nil {
```

Comment:
When an error (or any other value returned from a function) is not used _except_ for inside a single clause of an `if` statement immediately following it, you should declare it inside the `if` statement scope, e.g.,:

```go
if err := p.Match.UnmarshalBinary(data[n:]); err != nil {
        return err
}
```

This pattern is used everywhere in Go codebases.  It's better to shadow an existing variable (the function-level `err`) than to change its value for no reason.

In this function, there shouldn't even be a function-level `err` variable defined.  The `if` statement at the top of the function should also be following this inline assignment pattern.  I see a bunch of other functions which are also assigning function-level errors and not checking them, too, but these are problems in the existing code and can be fixed later. 😢